app/helpers/react_on_rails_helper.rb

```diff
@@ -224,6 +225,17 @@ def server_render_js(js_expression, options = {})
     # rubocop:enable Style/RaiseArgs
   end
 
+  def json_safe_and_pretty(hash_or_string)
+    unless hash_or_string.class.in?([Hash, String])
+      raise "#{__method__} only accepts String or Hash as argument "\
+            "(#{hash_or_string.class} given)."
+    end
+
+    json_value = hash_or_string.is_a?(String) ? hash_or_string : hash_or_string.to_json
+
+    ReactOnRails::JsonOutput.new(json_value).escaped
```

Comment:
This is where the performance might suffer because a new `JsonOutput` object is instantiated each time. Rather, `JsonOutput.escape_json(...)`, with `escape_json` being a class method can avoid this instantiation.